GitCommands/Git/GitModule.cs

```diff
@@ -1191,6 +1191,11 @@ public GitRevision GetRevision(string commit, bool shortFormat = false)
             return revision;
         }
 
+        public string GetCommitGpgStatus(string commit)
```

Comment:
Do you think you could extract all your new functionality into something like `GpgController` or `GpgManager` and inject `IGitModule`.
This way all your functionality is contained, it can be tested in isolation, and it will allow you to write unit tests 😀 
Also please make your methods `async`.
And you can drop "gpg" from method names too.

Your module then can be injected as a dependency where required, and ultimately will allow us to have less coupling and more testability.

You can have a look at how `GitRemoteController` is implemented and used.